apps/web-client/src/lib/stores/inventory/table_content.ts

```diff
@@ -3,43 +3,81 @@ import { debug } from '@librocco/shared';
 
 import type { NoteInterface, WarehouseInterface } from '@librocco/db';
 
-import type { BookStore, DisplayRow, PaginationData } from '$lib/types/inventory';
+import type { PaginationData, FullTableRow } from '$lib/types/inventory';
 
 import { readableFromStream } from '$lib/utils/streams';
+import { from, map, Observable, switchMap, tap } from 'rxjs';
+import { db } from '$lib/db';
+import type { DebugCtx } from '@librocco/shared/dist/debugger';
 
 interface CreateDisplayEntriesStore {
 	(
 		entity: NoteInterface | WarehouseInterface | undefined,
 		currentPageStore: Readable<number>,
-		bookStore: Readable<BookStore>,
 		ctx: debug.DebugCtx
-	): Readable<DisplayRow[]>;
+	): Readable<FullTableRow[]>;
 }
+
+interface createDisplayRowStream {
+	(entity: NoteInterface<object> | WarehouseInterface<NoteInterface<object>, object>, ctx: DebugCtx): Observable<
+		FullTableRow[]
+	>;
+}
+
+const createDisplayRowStream: createDisplayRowStream = (entity, ctx) => {
+	const fullTableRow = entity?.stream(ctx).entries.pipe(
+		switchMap((valueFromEntryStream) => {
+			const isbns = valueFromEntryStream.map((entry) => entry.isbn);
+			// map entry to just isbns
+			return from(db.getBooks(isbns)).pipe(
+				tap((val) => console.log({ val })),
+
+				map((booksFromDb) => {
+					// for each value from entry stream (volume stock client)
+					return valueFromEntryStream.map((individualEntry) => {
+						// find the corresponding book in booksFromDb
+						// instead of find => look in the same index then look at first element (maybe book is not in yet???)
+						const bookData = booksFromDb.find((book) => book.isbn === individualEntry.isbn);
+						// account for books not found
+						if (!bookData) return;
+						// return bookData + entry value
```

Comment:
If no book data found, we should return the (`VolumeStockClient`) entry (without book data)